buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/fixture/hadoop/tasks/AbstractClusterTask.groovy

```diff
@@ -22,11 +22,15 @@ package org.elasticsearch.hadoop.gradle.fixture.hadoop.tasks
 import org.elasticsearch.gradle.testclusters.DefaultTestClustersTask
 import org.elasticsearch.hadoop.gradle.fixture.hadoop.conf.HadoopClusterConfiguration
 import org.elasticsearch.hadoop.gradle.fixture.hadoop.conf.InstanceConfiguration
+import org.gradle.api.tasks.Internal
 
 abstract class AbstractClusterTask extends DefaultTestClustersTask {
 
+    @Internal
     HadoopClusterConfiguration clusterConfiguration
+    @Internal
     InstanceConfiguration executedOn
+    @Internal
```

Comment:
`@Internal` is used to indicate that the marked task property changing shouldn't cause a task to be rerun, but these all seem to be pretty important items - if the environment changes we should probably run. I'm not sure on the cluster config or which instance to execute from, but my gut says those as well. Should these all be `@Input` instead?

Reply:
Hmm maybe I misread the documentation for that annotation. You're probably right. I thought that if it was `@Internal` it would treat it as something that needed to be computed every time. I had intended to revisit it to see if I could use `@Input` here to speed things up if nothing had changed.

Reply:
Yeah I had misread this. Switched them over to @Input (except for a couple that are never actually changed as far as I can tell).